### PR TITLE
Merge editorialized release notes for version 17.6 into develop

### PR DESCRIPTION
**Pull request by mokagio**
This also includes 735cfd7e63, which didn't make it into the beta 1 merge because I noticed the build deployment CI failure it fixes only after opening that PR.

Nothing to test because these files don't affect the build.

## Regression Notes
1. Potential unintended areas of impact
N.A.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ ALL_BUNDLE_IDENTIFIERS = [
   'org.wordpress.WordPressAllTimeWidget',
   'org.wordpress.WordPressThisWeekWidget',
   'org.wordpress.WordPressIntents'
-].freeze
+]
```
Comment (mokagio) [inline fastlane/Fastfile:37]: This is a fix related to the ASC deployment which didn't make it into `develop` with beta 1.

**Review by mokagio — COMMENTED** [review]
- `release_notes.txt` updated as per editorial
- `AppStoreStrings.po` `msgctxt` updated to new version number
- `AppStoreStrings.po` `msgid` updated with formatted notes

**Comment by mokagio** [comment]
I mentioned in #16571 that we might need to add logic to generate a `.po[t]` file for the Jetpack release notes. I don't think that should block this PR though. The sooner we get this in the more chances we'll have of having the translations in in time.

File: WordPress/Resources/release_notes.txt [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+
+You’ve never been able to edit Post Archive pages because they’re automatically generated. In 17.6, you won’t be able to open them in the editor, either, saving you a wasted trip.
+
+We’ve also been working on fixing things:
+
+We took care of some notices that had been overlapping the ActionSheet, including one that displays Site Icon controls, another displaying the QuickStart Removal, and a third that displays More Actions in the editor.
+
+Last up, we squashed a bug that sometimes caused the app to crash during comment moderation and another to fix a login error for WordPress.org sites to show inline.
+
+That’s all for 17.6. See you in 17.7!
```
Comment (AliSoftware) [inline WordPress/Resources/release_notes.txt:17]: "That's all"? That's already a lot of things in a single release! 😄

**Review by AliSoftware — APPROVED** [review]
Diff LGTM, :shipit: 

---

As what you mention for for Jetpack Release notes, I agree we need to look into it – probably @jkmassel and @JavonDavis are the ones who will handle automation for this – though for now we will not really have JP-specific `RELEASE-NOTES.txt` files, see p1623854949162600-slack-C0180B5PRJ4 for details and discussion.

We will still likely need a `Jetpack/Resources/release_notes.txt` file and ask Editorial for a modified version of the notes for Jetpack, that will be based on the WP notes but omits anything Reader-related, and we will also need our scripts to generate the po file for Jetpack from those indeed.

**Pull request merged**